Apps/W1/HybridGP/app/src/Migration/Support/CheckBooks/GPCheckbookTransactions.table.al

```diff
@@ -216,8 +216,132 @@ table 40101 "GP Checkbook Transactions"
         }
     }
 
-    procedure MoveStagingData(CheckbookId: Code[15]; PostingGroup: Code[20])
+
+    procedure MoveStagingData(BankAccountNo: Code[20]; BankAccPostingGroupCode: Code[20]; CheckbookID: Text[15])
+    var
+        GenJournalLine: Record "Gen. Journal Line";
+        GenJournalBatch: Record "Gen. Journal Batch";
+        DataMigrationFacadeHelper: Codeunit "Data Migration Facade Helper";
+        AccountNo: Code[20];
+    begin
+        SetRange(CHEKBKID, CheckbookID);
+        if FindSet() then
+            repeat
+                if CMTrxType = 2 then begin
+                    AccountNo := GetBankAccPostingAccountNo(BankAccPostingGroupCode) /* GL Account Number based off of account index */
+                end else begin
+                    AccountNo := CMLinkID;  /* Vendor ID */
+                end;
+
+                CreateGeneralJournalLine(GenJournalLine,
+                    Format(CMRECNUM),
+                    DSCRIPTN,
+                    TRXDATE,
+                    AccountNo,
+                    TRXAMNT,
+                    BankAccountNo,
+                    CMTrxType
+                );
+
+            until Next() = 0;
+    end;
+
+    procedure CreateGeneralJournalLine(var GenJournalLine: Record "Gen. Journal Line"; DocumentNo: Code[20]; Description: Text[50]; PostingDate: Date; AccountNo: Code[20]; TrxAmount: Decimal; BankAccountNo: Code[20]; CMTrxType: Integer)
+    var
+        GenJournalLineCurrent: Record "Gen. Journal Line";
+        GenJournalTemplate: Record "Gen. Journal Template";
+        JournalTemplateName: Code[10];
+        AccountType: Enum "Gen. Journal Account Type";
+        DocumentType: Enum "Gen. Journal Document Type";
+        LineNum: Integer;
+    begin
+        CreateGeneralJournalBatchIfNeeded(CMTrxType);
+
+        /*  
+            GP CMTrxType we support
+            -- 2 = cash receipt
+            -- 3 = payment
+        */
+        if CMTrxType = 2 then begin
+            DocumentType := DocumentType::" ";
+            JournalTemplateName := 'CASHRCPT';
+            AccountType := AccountType::"G/L Account"
+            /* AccountNo will be GL Account number -- offset account */
+        end else begin
+            DocumentType := DocumentType::Payment;
+            JournalTemplateName := 'PAYMENT';
+            AccountType := AccountType::Vendor;
+            /* AccountNo will be Vendor account number */
+        end;
+
+        GenJournalLineCurrent.SetRange("Journal Template Name", JournalTemplateName);
+        GenJournalLineCurrent.SetRange("Journal Batch Name", 'GPBANK');
+        if GenJournalLineCurrent.FindLast then
+            LineNum := GenJournalLineCurrent."Line No." + 10000
```

Comment:
Does it have to increment by 10000, why not by just 1?

Reply:
that is the default value used within BC